actionpack/lib/action_controller/metal/conditional_get.rb

```diff
@@ -77,18 +77,16 @@ def etag(&etagger)
     #
     #   before_action { fresh_when @article, template: 'widgets/show' }
     #
-    def fresh_when(record_or_options, additional_options = {})
-      if record_or_options.is_a? Hash
```

Comment:
If you pass a hash as the first argument Ruby will use the hash as the keyword arguments because record defaults to nil.
See here:

```bash
irb(main):001:0> def old_fresh_when(record_or_options, additional_options = {})
irb(main):002:1>   puts record_or_options
irb(main):003:1>   puts additional_options
irb(main):004:1> end
=> :old_fresh_when
irb(main):005:0> old_fresh_when({})
{}
{}
=> nil
irb(main):006:0> def fresh_when(record = nil, **options)
irb(main):007:1>   puts record
irb(main):008:1>   puts options
irb(main):009:1> end
=> :fresh_when
irb(main):010:0> fresh_when(etag: 'hello')
# this line is empty because record is nil.
{:etag=>"hello"}
```